gui/app.py

```diff
@@ -1,15 +1,25 @@
+import dataclasses
+import numpy as np
 from PyQt5 import QtGui
-from PyQt5.QtWidgets import QWidget, QApplication, QLabel, QVBoxLayout, QTabWidget
+from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout, QTabWidget
 from PyQt5.QtGui import QPixmap
 import cv2
 from PyQt5.QtCore import pyqtSignal, pyqtSlot, Qt, QThread
-import numpy as np
 
 
-class Frame():
-    def __init__(self, cv_img, cam_index):
-        self.cv_img = cv_img
-        self.cam_index = cam_index
+@dataclasses.dataclass
+class Frame:
+    cv_img: np.ndarray
+    cam_index: int
+
+
+def convert_cv_qt(cv_img):
```

Comment:
There were some changes to this function to support greyscale images you'll want to merge in here.